packages/react/hooks.ts

```diff
@@ -232,11 +232,14 @@ export const useDeferredValue = (value: any) => {
 
 // useSyncExternalStore
 export const useSyncExternalStore = (subscribe, getSnapshot) => {
-  const state = useState(getSnapshot());
+  const [, updateState] = useState();
+  const forceUpdate = useCallback(() => updateState({}), []);
+  const [subscriber, setSubscriber] = useState(() => subscribe(forceUpdate));
+
   useEffect(() => {
-    subscribe(state);
-  });
-  subscribe(state);
+    if (subscriber) subscriber();
```

Comment:
Can be simplified and maybe save some bytes?
```suggestion
    subscriber?.();
```